h2quic/server.go

```diff
@@ -53,7 +53,8 @@ type Server struct {
 
 	supportedVersionsAsString string
 
-	pushEnabled map[protocol.ConnectionID]bool
+	pushEnabled       map[protocol.ConnectionID]bool
+	maxHeaderListSize map[protocol.ConnectionID]uint32
```

Comment:
Make this a single map and introduce a `{pushEnabled bool, maxHeaderListSize uint32}` struct.